File: src/main/java/pl/training/workshop/m6/s02_polymorphism/step2/PremiereScreening.java
package pl.training.workshop.m6.s02_polymorphism.step2;

import pl.training.workshop.shared.Money;

/** Krok 2: premiera - 30 minut spotkania z twórcami zamiast reklam. */
public final class PremiereScreening extends Screening {
    private final int runtime;

    PremiereScreening(String title, int runtime) {
        super(title);
        this.runtime = runtime;
    }

    @Override
    public String label() {
        return "Premiera: " + title();
    }

    @Override
    public int durationMinutes() {
        return 30 + runtime;
    }

    @Override
    public Money price() {
        return Money.of("35.00");
    }
}

File: csharp/src/Training.Workshop/M6/S02Polymorphism/Step2/PremiereScreening.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S02Polymorphism.Step2;

/// <summary>Krok 2: premiera - 30 minut spotkania z twórcami zamiast reklam.</summary>
public sealed class PremiereScreening : Screening
{
    private readonly int _runtime;

    internal PremiereScreening(string title, int runtime)
        : base(title)
    {
        _runtime = runtime;
    }

    public override string Label()
    {
        return "Premiera: " + Title;
    }

    public override int DurationMinutes()
    {
        return 30 + _runtime;
    }

    public override Money Price()
    {
        return Money.Of("35.00");
    }
}
